### PR TITLE
Issue 446: fixed bugs and small changes/enhancements

### PR DESCRIPTION
**Pull request by madhavkulkarni1986**

### CODE REVIEW

File: jarviscli/plugins/news.py [hunk 1]
```diff
@@ -123,8 +127,8 @@ def configure(self, jarvis):
         """
             configures the news channel of the user
         """
-        for index in self.source_dict.keys():
-            jarvis.say(str(index) + ": " + self.source_dict.get(index))
+        for index in sorted([ int(x) for x in self.source_dict.keys()]):
```
Comment (JonanOribe) [inline jarviscli/plugins/news.py]:
```suggestion
        for index in sorted([int(x) for x in self.source_dict.keys()]):
```

File: jarviscli/plugins/news.py [hunk 2]
```diff
@@ -165,31 +174,43 @@ def get_news(self, jarvis, searchlist):
...
+            else:
+                response = urllib.urlopen(url)
+        except urllib.error.HTTPError as err:
+            # Catch invalid key(Unauthorized) error
+            if err.code == 401:
+                jarvis.say("API key not valid", Fore.RED)
+                return None
+            # Catch some other errors
+            else:
+                jarvis.say("An error occured: Error code: "+ str(err.code), Fore.RED)
```
Comment (JonanOribe) [inline jarviscli/plugins/news.py]: jarvis.say("An error occured: Error code: " + str(err.code),

Comment (JonanOribe) [inline jarviscli/plugins/news.py]: jarvis.say("An error occured: Error code: " + str(err.code),

File: jarviscli/plugins/news.py [hunk 3]
```diff
@@ -200,11 +221,13 @@ def parse_articles(self, data, jarvis):
         # check if we have a valid index
         try:
             int(idx)
-            if int(idx) > index:
-                jarvis.say("Not a valid index")
+            if int(idx) > (index - 1):
+                jarvis.say(str(idx) + "is not a valid index",Fore.RED)
```
Comment (JonanOribe) [inline jarviscli/plugins/news.py]: jarvis.say(str(idx) + "is not a valid index", Fore.RED)

File: jarviscli/plugins/news.py [hunk 4]
```diff
@@ -200,11 +221,13 @@ def parse_articles(self, data, jarvis):
...
+                return
+            elif int(idx) == 0:
                 return
         except:
-            jarvis.say("Not a valid index")
+            jarvis.say("Not a valid index",Fore.RED)
```
Comment (JonanOribe) [inline jarviscli/plugins/news.py]: jarvis.say("Not a valid index", Fore.RED)

File: jarviscli/plugins/news.py (hunk 1 above)
Comment (madhavkulkarni1986) [reply jarviscli/plugins/news.py]: have fixed them already.. new build underway.. thanks for pointing out :)

**Comment by madhavkulkarni1986** [comment]
@appi147 Can you check and merge this PR please?

**Comment by appi147** [comment]
Looks good!! Let me test

**Comment by appi147** [comment]
closes #446

**Pull request merged**